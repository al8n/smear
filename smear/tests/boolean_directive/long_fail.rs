#[derive(smear::BooleanDirective)]
#[smear(
  long = "6n"
)]
struct StartWithNumber;

#[derive(smear::BooleanDirective)]
#[smear(
  long = "i"
)]
struct TooShort;


#[derive(smear::BooleanDirective)]
#[smear(
  long = ""
)]
struct EmptyString;

#[derive(smear::BooleanDirective)]
#[smear(
  long = "👍"
)]
struct Unicode;

#[derive(smear::BooleanDirective)]
#[smear(
  long = "invalid-character"
)]
struct InvalidCharacter;

fn main() {}