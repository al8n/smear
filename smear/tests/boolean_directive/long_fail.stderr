error: The first character of the `long` must be an ASCII alphabetic character or '_': got '6'.
 --> tests/boolean_directive/long_fail.rs:3:10
  |
3 |   long = "6n"
  |          ^^^^

error: The length of `long` must be larger than 1. If you want single character name, please set `short` instead.
 --> tests/boolean_directive/long_fail.rs:9:10
  |
9 |   long = "i"
  |          ^^^

error: The `long` cannot be empty.
  --> tests/boolean_directive/long_fail.rs:16:10
   |
16 |   long = ""
   |          ^^

error: The value of `long` contains an invalid character: '👍(👍:0)'.
  --> tests/boolean_directive/long_fail.rs:22:10
   |
22 |   long = "👍"
   |          ^^^^

error: The value of `long` contains an invalid character: 'invalid-character(-:7)'.
  --> tests/boolean_directive/long_fail.rs:28:10
   |
28 |   long = "invalid-character"
   |          ^^^^^^^^^^^^^^^^^^^
